src/lib/core/observe-content/observe-content.spec.ts

```diff
@@ -0,0 +1,63 @@
+import {Component} from '@angular/core';
+import {async, TestBed} from '@angular/core/testing';
+import {MutationObserverModule} from './observe-content';
+
+describe('Observe content', () => {
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [MutationObserverModule],
+      declarations: [ComponentWithTextContent, ComponentWithChildTextContent],
+    });
+
+    TestBed.compileComponents();
+  }));
+
+  describe('text content change', () => {
+    it('should call the registered for changes function', done => {
+      let fixture = TestBed.createComponent(ComponentWithTextContent);
+      fixture.detectChanges();
+
+      // If the hint label is empty, expect no label.
+      const spy = spyOn(fixture.componentInstance, 'doSomething').and.callFake(() => {
+        expect(spy.calls.any()).toBe(true);
+        done();
+      });
+
+      expect(spy.calls.any()).toBe(false);
+
+      fixture.componentInstance.text = 'text';
+      fixture.detectChanges();
+    });
```

Comment:
Just tried this locally and I saw that the async stuff doesn't really work. Seems like zones doesn't play well with `MutationObserver`...